pkg/service/model_build_target_group.go

```diff
@@ -483,3 +475,35 @@ func (t *defaultModelBuildTask) buildTargetGroupBindingNodeSelector(_ context.Co
 		MatchLabels: targetNodeLabels,
 	}, nil
 }
+
+func (t *defaultModelBuildTask) buildHealthCheckNetworkingIngressRules(trafficSource, hcSource []elbv2model.NetworkingPeer, tgPort, hcPort intstr.IntOrString,
+	tgProtocol corev1.Protocol, preserveClientIP, customSoureRanges bool) []elbv2model.NetworkingIngressRule {
+	if tgProtocol != corev1.ProtocolUDP &&
+		(hcPort.String() == healthCheckPortTrafficPort || hcPort.IntValue() == tgPort.IntValue()) {
+		if !preserveClientIP {
+			return []elbv2model.NetworkingIngressRule{}
+		}
+		if !customSoureRanges {
```

Comment:
LGTM for this version.
however, this becomes really complicated. we should refactor this logic here by some thing like compare CIDRs in the future.

Reply:
once we have SG support for NLB, this wouldn't be a matter of concern.